src/modules/formBuilder/components/itemEditor/pickLists/ManagePickListOptions.tsx

```diff
@@ -0,0 +1,104 @@
+import PickListOption from './PickListOption';
+import ControlledSelect from '@/modules/form/components/rhf/ControlledSelect';
+import { localResolvePickList } from '@/modules/form/util/formUtil';
+import CardGroup, {
+  RemovableCard,
+} from '@/modules/formBuilder/components/itemEditor/conditionals/CardGroup';
+import { FormItemControl } from '@/modules/formBuilder/components/itemEditor/types';
+import { Component } from '@/types/gqlTypes';
+
+export interface ManagePickListOptionsProps {
+  control: FormItemControl;
+  formItemComponent: Component;
+}
+
+const ManagePickListOptions: React.FC<ManagePickListOptionsProps> = ({
+  control,
+  formItemComponent,
+}) => {
+  const pickListTypesPickList = localResolvePickList('PickListType') || [];
+
+  const { fields, append, remove, replace } = useFieldArray({
+    control,
+    name: 'pickListOptions',
+  });
+
+  const {
+    field: { onChange: onChangePickListReference },
+  } = useController({
+    control,
+    name: 'pickListReference',
+  });
+
+  // `fields` above contains the defaultValues, so we also useWatch to get recent updates
+  const fieldsWatch = useWatch({
+    control,
+    name: 'pickListOptions',
+  });
+
+  const setPickListInitialSelected = useCallback(
+    (code: string) => {
+      if (!fieldsWatch) return;
+      const newFields = fieldsWatch.map((field) => {
+        return { ...field, initialSelected: field?.code === code };
+      });
+      replace(newFields);
+    },
+    [replace, fieldsWatch]
+  );
+
+  const isCodeUnique = useCallback(
+    (code: string) => {
+      if (!fieldsWatch) return true;
+      return fieldsWatch.filter((field) => field?.code === code).length <= 1;
+    },
+    [fieldsWatch]
+  );
+
+  return (
+    <>
+      <CardGroup
+        onAddItem={() => {
+          append({}, { shouldFocus: false });
+          onChangePickListReference(null);
+        }}
+        addItemText='Add Choice'
+      >
+        {fields.map((option, index) => (
+          <RemovableCard
+            key={option.id}
+            onRemove={() => remove(index)}
+            removeTooltip={'Remove Choice'}
+          >
+            <PickListOption
+              code={
+                fieldsWatch && fieldsWatch[index]
+                  ? fieldsWatch[index]?.code
+                  : ''
+              }
+              control={control}
+              index={index}
+              formItemComponent={formItemComponent}
+              setPickListInitialSelected={setPickListInitialSelected}
+              isCodeUnique={isCodeUnique}
+            />
+          </RemovableCard>
+        ))}
+      </CardGroup>
+      <ControlledSelect
+        name='pickListReference'
+        control={control}
+        label='Or, use a reference list for choices'
+        placeholder='Select list'
+        options={pickListTypesPickList}
+        onChange={(value) => {
+          if (value) replace([]);
+        }}
+      />
```

Comment:
Passing on onChange to the controlled RHF component feels like a bit of an anti pattern.
Another way to do this would be to `watch` the `pickListReference` value, and call `replace([])` if it has a value. That's the pattern [we use elsewhere](https://github.com/greenriver/hmis-frontend/blob/9fe5e55a56903ac4f6e89f15c63be030c924d315/src/modules/formBuilder/components/itemEditor/FormEditorItemProperties.tsx#L139-L145), so maybe we should stick to it for consistency.

Reply:
Okay! This works for removing the pick list options when a pick list reference is chosen. It does not quite work for nulling out the `initialSelected` values when one option's `initialSelected` is checked. I think this is because calling `replace` inside the `useEffect` with a non-empty list causes a rerender, which triggers `useEffect` again, and it creates an infinite loop.

One workaround I thought of is to use a custom validator instead of calling `replace`, similar to the custom validator `isCodeUnique`. The downside of this is that it does less work for the user and instead forces them to fix the error themselves, but maybe this is better anyway?

I was shying away from `useEffect` because of https://react.dev/learn/you-might-not-need-an-effect. Since this is ultimately a response to a user-generated event, I thought it would be more of an antipattern to use useEffect, and passing an additional callback would be preferable. Maybe the name of the callback could be changed from `onChange` to something else? Let me know what you think!

Reply:
Hmm, ok. Maybe passing an onChange handler is okay. It doesn't accept a `value`, so in that sense it's pretty clear that it can't act as a controlled component. (Ugh, maybe naming them ControlledSelect etc was confusing...since they really masquerade as "uncontrolled" from the perspective of the caller...)

Using validate also seems like a good solution to me, even if the validation only happens on save.

Reply:
Ok, I've pushed a change implementing this as an `afterChange` callback.

But, since this is so much effort and introducing a new pattern all for the sake of a little checkbox that isn't even working correctly and we aren't sure if there is a use case for it... Maybe we should:
- Remove this `afterChange` callback from the controlled checkbox
- Change the [validation mode](https://github.com/greenriver/hmis-frontend/pull/850#discussion_r1681404064) back to onSubmit instead of onChange
- Remove any onChange behavior on the initialSelected checkbox
- Validate on submit that only one option has initialSelected checked
- And as a bonus, this would eliminate the barrier [here](https://github.com/greenriver/hmis-frontend/pull/850#discussion_r1681105701) to implementing an 'advanced' collapse panel. It's all connected lol